From 8b37305459999f775310d2789426fadeb868b4a9 Mon Sep 17 00:00:00 2001
From: krishnapriya kanagaraj <krishnapriya_kanagaraj@comcast.com>
Source: COMCAST
License: LICENSE-LGPL-2.1
Upstream-Status: Pending
Signed-off-by: Krishna Priya Kanagraj <krishnapriya_kanagaraj@Comcast.Com>
Date: Thu, 3 Sep 2020 10:00:30 +0000
Subject: [PATCH] XRE-16048 : Playing Event is not in order

Reason for change: The Playing event should be triggered after the candidates are exchanged between peers and the pipeline is transitied to PLAYING state
Test Procedure: ensure the playing event should be fired after the exchange of candidates.The webrtc streaming should be working fine.
Risks: Low
---
 .../platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp     | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
index 0c2f29001070..30a7c381da8d 100644
--- a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -1580,6 +1580,10 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
 
 void MediaPlayerPrivateGStreamer::processBufferingStats(GstMessage* message)
 {
+#if ENABLE(MEDIA_STREAM) && !GST_CHECK_VERSION(1, 16, 0)
+    if (m_streamPrivate) return;
+#endif
+
     m_buffering = true;
     gst_message_parse_buffering(message, &m_bufferingPercentage);
 
@@ -2200,7 +2204,7 @@ void MediaPlayerPrivateGStreamer::updateStates()
         if (m_currentState == GST_STATE_READY)
             m_readyState = MediaPlayer::HaveNothing;
         else if (m_currentState == GST_STATE_PAUSED) {
-            m_readyState = MediaPlayer::HaveEnoughData;
+            m_readyState = MediaPlayer::HaveCurrentData;
             m_paused = true;
         } else if (m_currentState == GST_STATE_PLAYING)
             m_paused = false;
-- 
2.14.2

